authenticate/state.go

```diff
@@ -52,6 +56,10 @@ func newAuthenticateState() *authenticateState {
 }
 
 func newAuthenticateStateFromConfig(cfg *config.Config) (*authenticateState, error) {
+	if err := ValidateOptions(cfg.Options); err != nil {
```

Comment:
nit: if we make this `err` function scoped we can remove the uninitizialized err further down